python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -799,10 +799,19 @@ def nth(self, n):
         """
         Return the nth row from each group.
         """
-        result = self.agg(lambda x: x.nth(n)).sort_index()
-        sizes = self.size().sort_index()
 
-        return result[sizes > n]
+        self.obj["__groupbynth_order__"] = range(0, len(self.obj))
+        result = self.obj.groupby(self.grouping.keys).agg(lambda x: x.nth(n))
```

Comment:
It would be helpful if we added a comment above this line explaining why we are doing a groupby again.

Reply:
Done 👍